scripts/installer_test.go

```diff
@@ -0,0 +1,340 @@
+	defer requireUninstall(t, ctx)
+
+	expectedEmail := fmt.Sprintf("operator.email: %s", email)
+	expectedWallet := fmt.Sprintf("operator.wallet: \"%s\"", walletAddr)
+	expectedAddr := fmt.Sprintf("contact.external-address: %s", publicAddr)
+
+	configStr := readConfigLines(t, ctx, configPath)
+
+	require.Contains(t, configStr, expectedEmail)
+	require.Contains(t, configStr, expectedWallet)
+	require.Contains(t, configStr, expectedAddr)
+}
+
+func install(t *testing.T, ctx *testcontext.Context, msiPath, installDir string, args ...string) {
+	log.Printf("installing from %s\n", msiPath)
+	logPath := ctx.File("log", "install.log")
+	args = append(append([]string{
+		"/i", msiPath,
+		"/log", logPath,
+	}, append(msiBaseArgs, "INSTALLFOLDER="+installDir)...), args...)
+	//}, msiBaseArgs...), args...)
+
+	installOut, err := exec.Command("msiexec", args...).CombinedOutput()
+	if !assert.NoError(t, err) {
+		installLogData, err := ioutil.ReadFile(logPath)
+		if assert.NoError(t, err) {
+			t.Logf("MSIExec log:\n============================\n%s", string(installLogData))
+		}
+		t.Logf("MSIExec output:\n============================\n%s", string(installOut))
+		t.Fatal()
+	}
+}
+
+func requireUninstall(t *testing.T, ctx *testcontext.Context) {
+	logPath := ctx.File("log", "uninstall.log")
+	uninstallOut, err := uninstall(logPath)
+	if !assert.NoError(t, err) {
+		uninstallLogData, err := ioutil.ReadFile(logPath)
+		if !assert.NoError(t, err) {
+			t.Logf("MSIExec log:\n============================\n%s\n", string(uninstallLogData))
+		}
+		t.Fatalf("MSIExec output:\n============================\n%s", string(uninstallOut))
+	}
+}
+
+func uninstall(logPath string) ([]byte, error) {
+	if err := stopServices("storagenode", "storagenode-updater"); err != nil {
+		return []byte{}, err
+	}
+
+	args := append([]string{"/uninstall", *msiPathFlag, "/log", logPath}, msiBaseArgs...)
+	return exec.Command("msiexec", args...).CombinedOutput()
+}
+
+func stopServices(names ...string) (err error) {
+	serviceMgr, err := mgr.Connect()
+	if err != nil {
+		return err
+	}
+
+	group := new(errgroup.Group)
+	for _, name := range names {
+		name := name
+		group.Go(func() (err error) {
+			service, err := serviceMgr.OpenService(name)
+			if err != nil {
+				return err
+			}
+			defer func() {
+				err = errs.Combine(err, service.Close())
+			}()
+
+			if _, err = service.Control(svc.Stop); err != nil {
+				return err
+			}
+			return waitForStop(service)
+		})
+	}
+
+	return group.Wait()
+}
+
+func waitForStop(service *mgr.Service) error {
+	ctx, _ := context.WithTimeout(context.Background(), 30*time.Second)
+	for {
+		status, err := service.Query()
+		if err != nil {
+			return err
+		}
+
+		if err := ctx.Err(); err != nil {
+			return err
+		}
+
+		switch status.State {
+		case svc.Stopped:
+			return nil
+		default:
+			time.Sleep(500 * time.Millisecond)
+		}
+	}
+}
+
+func readConfigLines(t *testing.T, ctx *testcontext.Context, configPath string) string {
```

Comment:
why do we need this whole processing? raw config.yaml should also work with `require.Contains(t, configStr, expectedEmail)`, right?

Reply:
Initially I was trying to unmarshal the config into struct and this is left over from that. I suppose the only thing that this does that we would want to keep doing is just to make sure it's not in a commented line but we can do that with a `trim` and a `hasPrefix`.

Reply:
you may try https://github.com/spf13/viper#reading-config-from-ioreader. It might be a little bit easier.

Reply:
Oh cool... I didn't see how to do this without having a command object. Thanks!

Reply:
Using viper now